### ISSUE TITLE
[feature request]: Make pruning configurable by plugins

### ISSUE THREAD

**Issue by thiskevinwang** [hashicorp/waypoint#4035]
@thiskevinwang @paladin-devops - since this has come up a couple of times with plugins, might be worth opening a feature request and we can figure out how to support something like configuring pruning via plugins!

_Originally posted by @briancain in https://github.com/hashicorp/waypoint/issues/3941#issuecomment-1276252793_

**Comment by evanphx** [comment]
We need to do a good write up about mutable deployments play with releasers, so that we can get a good understanding for users and also, if there is a bug/undesirable behavior that we want to fix.

**Comment by streple** [comment]
This would be awesome for the nomad jobspec releaser. 
Currently git pooling is unusable together with nomad jobspec as I tried to describe  [here](https://discuss.hashicorp.com/t/every-deployment-ends-with-no-allocations-for-job-are-running-third-deployment-prunes-the-job-from-nomad-away/45089/7) .

**Comment by thiskevinwang** [comment]
@evanphx two questions:
1. Do _mutable deployments_ exist today?
2. With _mutable deployments_, is the idea that each `waypoint up` continuously updates a single `v1` deployment (as opposed to accumulating `v1`, `v2`, `v3`, ...etc. ?

**Comment by evanphx** [comment]
@thiskevinwang Yup, they exist and they work as you stated, by updating a single deployment.
